### DISCUSSION TITLE
(ANSWERED) twilight times

### DISCUSSION THREAD

**Discussion by calders** [skyfielders/python-skyfield#597] [Q&A] [answered]
Hi,

I tried the ["When will it get dark tonight?"](https://rhodesmill.org/skyfield/examples.html#dark-twilight-day-example) example for my own location. I changed the following lines:

```
from skyfield.api import N, W, E, wgs84, load # adding E
uccle = wgs84.latlon(50.7986 * N, 4.3581 * E)
f = almanac.dark_twilight_day(eph, uccle)
```

When I try to run it, I get the following error:
```
Traceback (most recent call last):

  File "C:\Users\stijn\Desktop\twilight.py", line 35, in <module>
    print(tstr, ' ', almanac.TWILIGHTS[previous_e], 'ends')

TypeError: unhashable type: 'numpy.ndarray'
```

What do I do wrong? Thanks for helping me!

**Comment by JoshPaterson** [comment, marked as answer]
Hi! Here's how I edited the script to make it work:
```
import datetime as dt
from pytz import timezone
from skyfield import almanac
from skyfield.api import N, E, wgs84, load
import numpy as np

# Figure out local midnight.
zone = timezone('US/Eastern')
now = zone.localize(dt.datetime.now())
midnight = now.replace(hour=0, minute=0, second=0, microsecond=0)
next_midnight = midnight + dt.timedelta(days=1)

ts = load.timescale()
t0 = ts.from_datetime(midnight)
t1 = ts.from_datetime(next_midnight)
eph = load('de421.bsp')
uccle = wgs84.latlon(50.7986 * N, 4.3581 * E)
f = almanac.dark_twilight_day(eph, uccle)
times, events = almanac.find_discrete(t0, t1, f)

previous_e = f(t0).item()
for t, e in zip(times, events):
    tstr = str(t.astimezone(zone))[:16]
    if previous_e < e:
        print(tstr, ' ', almanac.TWILIGHTS[e], 'starts')
    else:
        print(tstr, ' ', almanac.TWILIGHTS[previous_e], 'ends')
    previous_e = e
```
I too got the same error as you did. I fixed it by adding `.item()` where `previous_e` is defined. I think this is an error in the example script, so I created a pull request to fix it!

**Reply by calders** [reply]
Thanks! This solution works!

**Reply by brandon-rhodes** [reply]
I've applied Josh's patch, so this tweak should also appear in the docs the next time Skyfield is released. Thanks, Josh!

**Reply by HeySmarty** [reply]
Hello, if someone belongs to a timezone Asia/Kolkata which is GMT + 05:30, how should the following line needs to be modified:
tstr = str(t.astimezone(zone))[:16]

Thanks in advance.
